python/lsst/skymap/packers.py

```diff
@@ -19,9 +19,47 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+from __future__ import annotations
+
 __all__ = ("SkyMapDimensionPacker",)
 
+from collections.abc import Mapping
+
+from lsst.pex.config import Config, Field, DictField, ConfigurableField
 from lsst.daf.butler import DimensionPacker, DimensionGraph, DataCoordinate
+from deprecated.sphinx import deprecated
+
+
+class SkyMapDimensionPackerConfig(Config):
+    bands = DictField(
+        "Mapping from band name to integer to use in the packed ID.",
+        keytype=str,
+        itemtype=int,
+        default=None,
+        optional=True,
```

Comment:
This is optional and default=None, meaning it will use the default list from below. Should it have `default={}` instead, and/or be non-optional, since the goal is to ensure that new code doesn't rely on that hard-coded list?

Alternately, the docstring should probably mention what passing `None` does.

Reply:
I've walked back the admonition against letting this stay at the default a bit - I think that default is important to keep this ticket from being more disruptive than it needs to be, even though I think it is preferable for pipelines to define their own when possible.